src/Cake.Common.Tests/Unit/IO/DirectoryAliasesTests.cs

```diff
@@ -4,6 +4,7 @@
 using Cake.Common.Tests.Fixtures;
 using Cake.Core;
 using Cake.Core.IO;
+using Cake.Core.Tests.Fakes;
```

Comment:
Hi @RichiCoder1,

I got it to compile by changing usings to
```csharp
using System;
using System.Collections.Generic;
using System.IO;
using Cake.Common.IO;
using Cake.Common.Tests.Fixtures;
using Cake.Core;
using Cake.Core.IO;
using Cake.Testing.Fakes;
using NSubstitute;
using Xunit;
```
And adjusted your tests to
```csharp
        public sealed class TheCopyDirectoryMethod
        {
            [Fact]
            public void Should_Throw_If_Context_Is_Null()
            {
                // Given
                var sourcePath = new DirectoryPath("C:/Temp");
                var destinationPath = new DirectoryPath("C:/Temp2");

                // When
                var result = Record.Exception(() =>
                    DirectoryAliases.CopyDirectory(null, sourcePath, destinationPath));

                // Then
                Assert.IsArgumentNullException(result, "context");
            }

            [Fact]
            public void Should_Throw_If_Source_Directory_Is_Null()
            {
                // Given
                var context = Substitute.For<ICakeContext>();

                // When
                var result = Record.Exception(() =>
                    context.CopyDirectory(null, null));

                // Then
                Assert.IsArgumentNullException(result, "source");
            }

            [Fact]
            public void Should_Throw_If_Destination_Directory_Is_Null()
            {
                // Given
                var context = Substitute.For<ICakeContext>();
                var sourcePath = new DirectoryPath("C:/Temp");

                // When
                var result = Record.Exception(() =>
                    context.CopyDirectory(sourcePath, null));

                // Then
                Assert.IsArgumentNullException(result, "destination");
            }

            [Fact]
            public void Should_Throw_If_Source_Directory_Does_Not_Exist()
            {
                // Given
                var context = Substitute.For<ICakeContext>();
                context.FileSystem.Returns(new FakeFileSystem(false));
                var sourcePath = new DirectoryPath("C:/Temp");
                var destinationPath = new DirectoryPath("C:/Temp2");

                // When
                var result = Record.Exception(() =>
                    context.CopyDirectory(sourcePath, destinationPath));

                // Then
                Assert.IsType<DirectoryNotFoundException>(result);
            }

            [Fact]
            public void Should_Create_Destination_Folder_If_Not_Exist()
            {
                // Given
                var context = Substitute.For<ICakeContext>();
                var fileSystem = CreateFileStructure(new FakeFileSystem(false));
                context.FileSystem.Returns(fileSystem);
                var sourcePath = new DirectoryPath("C:/Temp");
                var destinationPath = new DirectoryPath("C:/Temp2");

                // When
                context.CopyDirectory(sourcePath, destinationPath);
                
                // Then
                Assert.True(fileSystem.Directories.ContainsKey(destinationPath));
            }

            [Fact]
            public void Should_Copy_Files()
            {
                // Given
                var context = Substitute.For<ICakeContext>();
                var fileSystem = CreateFileStructure(new FakeFileSystem(false));
                context.FileSystem.Returns(fileSystem);
                var sourcePath = new DirectoryPath("C:/Temp");
                var destinationPath = new DirectoryPath("C:/Temp2");

                // When
                context.CopyDirectory(sourcePath, destinationPath);

                // Then
                Assert.True(fileSystem.Files.ContainsKey(new FilePath("C:/Temp/file1.txt")));
                Assert.True(fileSystem.Files.ContainsKey(new FilePath("C:/Temp/file2.txt")));
            }

            [Fact]
            public void Should_Recursively_Copy_Files_And_Directory()
            {
                // Given
                var context = Substitute.For<ICakeContext>();
                var fileSystem = CreateFileStructure(new FakeFileSystem(false));
                context.FileSystem.Returns(fileSystem);
                var sourcePath = new DirectoryPath("C:/Temp");
                var destinationPath = new DirectoryPath("C:/Temp2");

                // When
                context.CopyDirectory(sourcePath, destinationPath);

                // Then
                // Directories should exist
                Assert.True(fileSystem.Directories.ContainsKey(new DirectoryPath("C:/Temp2/Stuff")));
                Assert.True(fileSystem.Directories.ContainsKey(new DirectoryPath("C:/Temp2/Things")));

                // Files should exist
                Assert.True(fileSystem.Files.ContainsKey(new FilePath("C:/Temp2/Stuff/file1.txt")));
                Assert.True(fileSystem.Files.ContainsKey(new FilePath("C:/Temp2/Stuff/file2.txt")));
                Assert.True(fileSystem.Files.ContainsKey(new FilePath("C:/Temp2/Things/file1.txt")));
            }

            private FakeFileSystem CreateFileStructure(FakeFileSystem ffs)
            {
                Action<string> dir = (path) => ffs.GetCreatedDirectory(path);
                Action<string> file = (path) => ffs.GetCreatedFile(path);

                dir("C:/Temp");
                {
                    file("C:/Temp/file1.txt");
                    file("C:/Temp/file2.txt");
                    dir("C:/Temp/Stuff");
                    {
                        file("C:/Temp/Stuff/file1.txt");
                        file("C:/Temp/Stuff/file2.txt");
                    }
                    dir("C:/Temp/Things");
                    {
                        file("C:/Temp/Things/file1.txt");
                    }
                }

                return ffs;
            }
        }
```
It then compiles but the tests fail
```
Starting:    Cake.Common.Tests
   Cake.Common.Tests.Unit.IO.DirectoryAliasesTests+TheCopyDirectoryMethod.Should_Copy_Files [FAIL]
      System.InvalidOperationException : Cannot combine a directory path with an absolute directory path.
      Stack Trace:
         src\Cake.Core\IO\DirectoryPath.cs(77,0): at Cake.Core.IO.DirectoryPath.Combine(DirectoryPath path)
         src\Cake.Common\IO\DirectoryAliases.cs(211,0): at Cake.Common.IO.DirectoryAliases.CopyDirectory(ICakeContext context, DirectoryPath source, DirectoryPath destinati
on)
         src\Cake.Common.Tests\Unit\IO\DirectoryAliasesTests.cs(832,0): at Cake.Common.Tests.Unit.IO.DirectoryAliasesTests.TheCopyDirectoryMethod.Should_Copy_Files()
   Cake.Common.Tests.Unit.IO.DirectoryAliasesTests+TheCopyDirectoryMethod.Should_Recursively_Copy_Files_And_Directory [FAIL]
      System.InvalidOperationException : Cannot combine a directory path with an absolute directory path.
      Stack Trace:
         src\Cake.Core\IO\DirectoryPath.cs(77,0): at Cake.Core.IO.DirectoryPath.Combine(DirectoryPath path)
         src\Cake.Common\IO\DirectoryAliases.cs(211,0): at Cake.Common.IO.DirectoryAliases.CopyDirectory(ICakeContext context, DirectoryPath source, DirectoryPath destinati
on)
         src\Cake.Common.Tests\Unit\IO\DirectoryAliasesTests.cs(850,0): at Cake.Common.Tests.Unit.IO.DirectoryAliasesTests.TheCopyDirectoryMethod.Should_Recursively_Copy_Fi
les_And_Directory()
   Cake.Common.Tests.Unit.IO.DirectoryAliasesTests+TheCopyDirectoryMethod.Should_Create_Destination_Folder_If_Not_Exist [FAIL]
      System.InvalidOperationException : Cannot combine a directory path with an absolute directory path.
      Stack Trace:
         src\Cake.Core\IO\DirectoryPath.cs(77,0): at Cake.Core.IO.DirectoryPath.Combine(DirectoryPath path)
         src\Cake.Common\IO\DirectoryAliases.cs(211,0): at Cake.Common.IO.DirectoryAliases.CopyDirectory(ICakeContext context, DirectoryPath source, DirectoryPath destinati
on)
         src\Cake.Common.Tests\Unit\IO\DirectoryAliasesTests.cs(815,0): at Cake.Common.Tests.Unit.IO.DirectoryAliasesTests.TheCopyDirectoryMethod.Should_Create_Destination_
Folder_If_Not_Exist()
Finished:    Cake.Common.Tests
```
So there seems to be some issue with the implementation or tests, hope this info helps you pin point/resolve the issue.
Regards,
/Mattias